internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -2062,8 +2062,8 @@ func resourceKubernetesClusterUpdate(d *pluginsdk.ResourceData, meta interface{}
 			}
 		}
 
-		// if the default node pool name has changed it means the initial attempt at resizing failed
-		if d.HasChange("default_node_pool.0.vm_size") || d.HasChange("default_node_pool.0.name") {
+		// if the default node pool name has changed, it means the initial attempt at resizing failed
```

Comment:
The fields `os_sku`, `os_disk_size_gb` and `os_disk_type` will need to have `ForceNew` removed in the schema.